manifest/discovery/discovery.go

```diff
@@ -0,0 +1,289 @@
+func (d Driver) InDevelopment() bool {
+	return d.Status.Rank() < manifest.Alpha.Rank()
+}
+
+// IsRecommended indicates that driver is stable enough to be used in production.
+func (d Driver) IsRecommended() bool {
+	return d.Status.Rank() >= manifest.Beta.Rank()
+}
+
+// RepositoryURL returns Github repository URL for browsers (not git).
+func (d Driver) RepositoryURL() string {
+	return d.repo.GetHTMLURL()
+}
+
+// repositoryFileURL returns an URL of file in the driver's repository.
+func (d Driver) repositoryFileURL(path string) string {
+	path = strings.TrimPrefix(path, "/")
+	return fmt.Sprintf("https://raw.githubusercontent.com/%s/master/%s", d.repo.GetFullName(), path)
+}
+
+// newReq constructs a GET request with context.
+func newReq(ctx context.Context, url string) *http.Request {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		// if it fails, it's a programmer's error
+		panic(err)
+	}
+	return req.WithContext(ctx)
+}
+
+// loadManifest reads manifest file from repository and decodes it into object.
+func (d *Driver) loadManifest(ctx context.Context) error {
+	req := newReq(ctx, d.repositoryFileURL(manifest.Filename))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		// outdated driver
+		d.Name = d.Language
+		d.Status = manifest.Inactive
+		return nil
+	} else if resp.StatusCode/100 != 2 { // 2xx
+		return fmt.Errorf("status: %v", resp.Status)
+	}
+
+	lang := d.Language
+	if err := d.Manifest.Decode(resp.Body); err != nil {
+		return err
+	}
+	// override language ID from manifest (prevents copy-paste of manifests)
+	d.Language = lang
+	if d.Name == "" {
+		d.Name = d.Language
+	}
+	return nil
+}
+
+// reMaintainer is a regexp for one line of MAINTAINERS file (Github handle is optional):
+//
+//		John Doe <john@domain.com> (@john_at_github)
+var reMaintainer = regexp.MustCompile(`^([^<(]+)\s<([^>]+)>(\s\(@([^\s]+)\))?`)
+
+// loadMaintainers reads MAINTAINERS file from repository and decodes it into object.
+//
+// Each line in a file should follow the format defined by reMaintainer regexp.
+func (d *Driver) loadMaintainers(ctx context.Context) error {
+	req := newReq(ctx, d.repositoryFileURL("MAINTAINERS"))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusNotFound {
+		return nil
+	} else if resp.StatusCode/100 != 2 {
+		return fmt.Errorf("status: %v", resp.Status)
+	}
+
+	var out []Maintainer
+	sc := bufio.NewScanner(resp.Body)
+	for sc.Scan() {
+		line := strings.TrimSpace(sc.Text())
+		sub := reMaintainer.FindStringSubmatch(line)
+		if len(sub) == 0 {
+			continue
+		}
+		m := Maintainer{Name: sub[1], Email: sub[2]}
+		if len(sub) >= 5 {
+			m.Github = sub[4]
+		}
+		out = append(out, m)
+	}
+	d.Maintainers = out
+	return nil
+}
+
```

Comment:
Should have a comment here, and above the options to have the godoc correctly generated.

Reply:
Options fields are OK as-is (godoc supports both inline and header comments), but thanks for spotting missing docs for Options itself.